src/components/SignUpForm/index.tsx

```diff
@@ -0,0 +1,230 @@
+import React, { useState } from 'react';
+
+import { useNavigate } from 'react-router-dom';
+import ArrowBackIcon from '@mui/icons-material/ArrowBack';
+import axios from 'axios';
+import type { AxiosError } from 'axios';
+import type { ChangeEvent } from 'react';
+import type { SelectChangeEvent } from '@mui/material';
+import {
+  FormControl,
+  FormControlLabel,
+  FormHelperText,
+  FormLabel,
+  InputLabel,
+  MenuItem,
+  Radio,
+  RadioGroup,
+  Select,
+  TextField,
+  Typography,
+} from '@mui/material';
+
+import { validateString } from 'utils/validate-string';
+
+import { Main, Tips, ButtonsContainer, StyledButton } from './styles';
+
+const TIME_TO_REDIRECT = 6000;
+const INITIAL_DATA = {
+  name: '',
+  lastname: '',
+  nickname: '',
+  email: '',
+  gender: '',
+  expertise: '',
+};
+
+interface ISignUpForm {
+  setSnackbar: (
+    value: React.SetStateAction<{
+      open: boolean;
+      message: string;
+      severity: string;
+    }>,
+  ) => void;
+}
+
+const SignUpForm: React.FC<ISignUpForm> = ({ setSnackbar }) => {
+  const navigate = useNavigate();
+
+  const [loading, setLoading] = useState(false);
+  const [formData, setFormData] = useState(INITIAL_DATA);
+
+  const handleChange = (event: ChangeEvent<HTMLInputElement>): void => {
+    setFormData(curr => ({
+      ...curr,
+      [event.target.name]: event.target.value || '',
+    }));
+  };
+
+  const handleChangeSelect = (event: SelectChangeEvent): void => {
+    setFormData(curr => ({
+      ...curr,
+      [event.target.name]: event.target.value || undefined,
+    }));
+  };
+
+  const handleSubmit = async (): Promise<void> => {
+    setLoading(true);
+    try {
```

Comment:
You can do it with then().error().finally()

Reply:
No, I can't in this case. `validateString` throws errors that I need to catch before request is made